tests/test_builder.py

```diff
@@ -6,45 +9,95 @@
 @pytest.fixture
 def gb():
     b = Builder("./example/create_gui.yaml")
-    b.setup()
     return b
 
 
-def test_guibuilder(gb: Builder):
-    assert gb.beamline.dom == "bl01t"
-    assert gb.beamline.desc == "Test Beamline"
-    assert gb.components[0].name == "fshtr"
-    assert gb.components[0].desc == "Fast Shutter"
-    assert gb.components[0].P == "BL01T-EA-FSHTR-01"
-    assert gb.components[0].R is None
-    assert gb.components[0].attribute is None
-
-
-def test_gb_extract_services(gb):
-    gb._extract_services()
-
-
-#     assert gb.entities[0].type == "pmac.autohome"
-#     assert gb.entities[0].DESC == "motor"
-#     assert gb.entities[0].P == "BL01T-MO-MAP-01:STAGE"
-#     assert gb.entities[0].M is None
-#     assert gb.entities[0].R is None
-#     assert gb.entities[1].type == "pmac.dls_pmac_asyn_motor"
-#     assert gb.entities[1].DESC == "motor"
-#     assert gb.entities[1].P == "BL01T-MO-MAP-01:STAGE"
-#     assert gb.entities[1].M == "X"
-#     assert gb.entities[1].R is None
-#     assert gb.entities[2].type == "pmac.dls_pmac_asyn_motor"
-#     assert gb.entities[2].DESC == "motor"
-#     assert gb.entities[2].P == "BL01T-MO-MAP-01:STAGE"
-#     assert gb.entities[2].M == "A"
-#     assert gb.entities[1].R is None
-
-
-# def test_gui_map(gb):
-#     gb.find_services_folders()
-#     gb.gui_map()
-#     assert os.path.isfile("./motor.bob")
-#     assert os.path.isfile("./map.json")
-
-# TODO Tests
+@pytest.mark.parametrize(
+    "attr, expected",
+    [
+        ("beamline.dom", "bl01t"),
+        ("beamline.desc", "Test Beamline"),
+    ],
+)
+def test_beamline_attributes(gb: Builder, attr, expected):
+    assert eval(f"gb.{attr}") == expected
+
+
+@pytest.mark.parametrize(
+    "index, name, desc, P, R, attribute, service_name",
+    [
+        (0, "fshtr", "Fast Shutter", "BL01T-EA-FSHTR-01", None, None, None),
+        (
+            4,
+            "motor",
+            "Hexapod Stage",
+            "BL01T-MO-MAP-01",
+            "STAGE",
+            None,
+            "bl01t-mo-ioc-01",
+        ),
+    ],
+)
+def test_component_attributes(
+    gb: Builder, index, name, desc, P, R, attribute, service_name
```

Comment:
Is it possible to add type-hints?

Reply:
Hmm we might not need them to be honest, it seems to be able to infer the types automatically from the parameters...